app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,63 @@
+class CommentsController < ApplicationController
+  before_action :set_comment, only: %i[show edit update destroy]
+  before_action :set_task, only: %i[new index create destroy]
+
+  def index
+    @comments = @task.comments
+  end
+
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+
+    if @comment.save
+      redirect_to project_task_comment_url(@task.project, @task, @comment), notice: "Comment was successfully created!"
+    else
+      flash.now[:notice] = "Something went wrong. Try again."
+      render :new
+    end
+  end
+
+  def destroy
+    @comment.destroy
+    redirect_to @task, notice: "Comment was successfully destroyed."
+  end
+
+  def show
+  end
+
+  def edit
+  end
+
+  def update
+    if @comment.update(comment_params)
+      redirect_to project_task_comment_url(@comment.task.project, @comment.task, @comment),
+                  notice: "Comment was successfully updated."
+    else
+      flash.now[:notice] = "Something went wrong. Try again."
+      render :edit
+    end
+  end
+
+  private
+
+  def set_comment
+    @comment = Comment.find(params[:id])
+    @task = @comment.task
+    @project = @task.project
+    authorize! @project, with: TaskPolicy
+  end
+
+  def set_task
+    @task = Task.find(params[:task_id])
+    @project = @task.project
+    authorize! @project, with: TaskPolicy
```

Comment:
Why? If only the project owner can create tasks, then only the project owner should be able to create comments?

Reply:
It doesn't work like that, my policy allows to create comments if user and project have any relation in project_membership

Reply:
Yes, I understand that. But you associate the rights to create a comment with the rights to create a task. If we change the logic of creating a task, it will also affect the logic of creating a comment.